Installation issue: gpgme

### Steps to reproduce the issue

```console
$ spack spec -I gpgme
Input spec
--------------------------------
 -   gpgme

Concretized
--------------------------------
 -   gpgme@1.20.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]      ^gcc-runtime@11.4.0%gcc@11.4.0 build_system=generic arch=linux-ubuntu22.04-westmere
[e]      ^glibc@2.35%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]      ^gmake@4.4.1%gcc@11.4.0~guile build_system=generic arch=linux-ubuntu22.04-westmere
[+]      ^gnupg@2.4.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]          ^libgcrypt@1.11.0%gcc@11.4.0 build_system=autotools patches=4e308ba arch=linux-ubuntu22.04-westmere
[+]          ^libiconv@1.17%gcc@11.4.0 build_system=autotools libs=shared,static arch=linux-ubuntu22.04-westmere
[+]          ^libksba@1.6.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]          ^npth@1.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]          ^pinentry@1.3.1%gcc@11.4.0 build_system=autotools gui=tty arch=linux-ubuntu22.04-westmere
[+]          ^zlib-ng@2.2.1%gcc@11.4.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=linux-ubuntu22.04-westmere
[+]      ^libassuan@3.0.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]      ^libgpg-error@1.50%gcc@11.4.0 build_system=autotools patches=0b2a0ff arch=linux-ubuntu22.04-westmere
[+]          ^gawk@5.3.0%gcc@11.4.0~nls build_system=autotools arch=linux-ubuntu22.04-westmere
[+]              ^gmp@6.3.0%clang@15.0.7+cxx build_system=autotools libs=shared,static arch=linux-ubuntu22.04-westmere
[+]                  ^autoconf@2.72%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                      ^perl@5.38.2%gcc@11.4.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu22.04-westmere
[+]                          ^berkeley-db@18.1.40%gcc@11.4.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-westmere
[+]                          ^bzip2@1.0.8%gcc@11.4.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-westmere
[+]                          ^gdbm@1.23%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                  ^automake@1.16.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                  ^libtool@2.4.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                      ^findutils@4.9.0%gcc@11.4.0 build_system=autotools patches=440b954 arch=linux-ubuntu22.04-westmere
[+]                  ^m4@1.4.19%gcc@11.4.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-westmere
[+]                      ^diffutils@3.10%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]              ^libsigsegv@2.14%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]              ^mpfr@4.2.1%gcc@11.4.0 build_system=autotools libs=shared,static arch=linux-ubuntu22.04-westmere
[+]                  ^autoconf-archive@2023.02.20%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                  ^texinfo@7.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                      ^gettext@0.22.5%gcc@11.4.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                          ^libxml2@2.10.3%gcc@11.4.0+pic~python+shared build_system=autotools arch=linux-ubuntu22.04-westmere
[+]                          ^tar@1.34%gcc@11.4.0 build_system=autotools zip=pigz arch=linux-ubuntu22.04-westmere
[+]                              ^pigz@2.8%gcc@11.4.0 build_system=makefile arch=linux-ubuntu22.04-westmere
[+]                              ^zstd@1.5.6%gcc@11.4.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu22.04-westmere
[+]                          ^xz@5.4.6%gcc@11.4.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu22.04-westmere
[+]              ^readline@8.2%gcc@11.4.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-westmere
[+]                  ^ncurses@6.5%gcc@11.4.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=linux-ubuntu22.04-westmere
[+]                      ^pkgconf@2.2.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu22.04-westmere

```


### Error message

<details><summary>Error message</summary>

<pre>
$ spack install gpgme
[+] /usr (external glibc-2.35-isjlnk4dwcubzx7hhhtixglpbtlkzypp)
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/gcc-runtime-11.4.0-nk6gexj57nozfiqcqflb6h7xuzjz3wxz
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/libgpg-error-1.50-p4a7eqbp2gkjs3qw6von7x6lh7ld74pk
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/zlib-ng-2.2.1-sf752leyvgvr7leb52igbard6tte6qcl
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/npth-1.7-ehcl3kcbnm7pmus4fq7efm6d4cu7cw4x
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/gmake-4.4.1-j3pgxcqx4vepmvwlyvbe5x2kaphv7bux
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/libksba-1.6.7-3xpgvioxxovupr4nvqc3q76ne5j7s54o
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/libgcrypt-1.11.0-cag7nlhzrhqt2yuowwvimahcsmp6yzqp
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/libassuan-3.0.1-4n4womos6ret3a4d5752nvqji4efuxdw
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/libiconv-1.17-vuefcnk5vyg2jqk2ywgesf2iasqxbp5m
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/pinentry-1.3.1-w26myuqza2tf3pfzm756lstnnswlaayw
[+] /home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/gnupg-2.4.5-o5glrezwzdmj3bonnk6ydcyfbrl4h25l
==> Installing gpgme-1.20.0-3d6werfk7nlr5gsghudslqhlq2rlzwea [13/13]
==> No binary for gpgme-1.20.0-3d6werfk7nlr5gsghudslqhlq2rlzwea found: installing from source
==> Using cached archive: /home/wspear/bin/SPACK/spack/var/spack/cache/_source-cache/archive/25/25a5785a5da356689001440926b94e967d02e13c49eb7743e35ef0cf22e42750.tar.bz2
==> No patches needed for gpgme
==> gpgme: Executing phase: 'autoreconf'
==> gpgme: Executing phase: 'configure'
==> Error: ProcessError: Command exited with status 1:
    '/tmp/wspear/spack-stage/spack-stage-gpgme-1.20.0-3d6werfk7nlr5gsghudslqhlq2rlzwea/spack-src/configure' '--prefix=/home/wspear/bin/SPACK/spack/opt/spack/linux-ubuntu22.04-westmere/gcc-11.4.0/gpgme-1.20.0-3d6werfk7nlr5gsghudslqhlq2rlzwea' '--enable-languages=cpp'

1 error found in build log:
     165    ***
     166    *** You need libassuan to build this program.
     167    *** This library is for example available at
     168    ***   https://www.gnupg.org/ftp/gcrypt/libassuan/
     169    *** (at least version 2.4.2 (API 2) is required).
     170    ***
  >> 171    configure: error:
     172    ***
     173    *** Required libraries not found. Please consult the above messages
     174    *** and install them before running configure again.
     175    ***

See build log for details:
  /tmp/wspear/spack-stage/spack-stage-gpgme-1.20.0-3d6werfk7nlr5gsghudslqhlq2rlzwea/spack-build-out.txt

</pre></details>


### Information on your system

* **Spack:** 0.23.0.dev0 (553b44473f0cb7d1210829099b9e0b968d6e354c)
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-westmere


### Additional information

Logs: http://yu.nic.uoregon.edu/~wspear/gpgme-fault/

Initially discovered while installing gpgme as a dependency of podman.

It looks like it can't find libassuan despite it being present as a dependency.

The error seems unrelated to this existing gpgme build issue: https://github.com/spack/spack/issues/12617

@baberlevi @alecbcs @paulbry  (No maintainers; I identified significant contributors from the git log)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/gpgme/package.py
+++ b/var/spack/repos/builtin/packages/gpgme/package.py
@@ -33,7 +33,7 @@ class Gpgme(AutotoolsPackage):
 
     depends_on("gnupg", type="build")
     depends_on("libgpg-error", type="build")
-    depends_on("libassuan@:2.5.7", type="build")
+    depends_on("libassuan@2.4.2:2", type="build")
 
     @classmethod
     def determine_version(cls, exe):

--- a/var/spack/repos/builtin/packages/gpgme/package.py
+++ b/var/spack/repos/builtin/packages/gpgme/package.py
@@ -33,7 +33,7 @@ class Gpgme(AutotoolsPackage):
 
     depends_on("gnupg", type="build")
     depends_on("libgpg-error", type="build")
-    depends_on("libassuan", type="build")
+    depends_on("libassuan@:2.5.7", type="build")
 
     @classmethod
     def determine_version(cls, exe):

--- a/var/spack/repos/builtin/packages/gpgme/package.py
+++ b/var/spack/repos/builtin/packages/gpgme/package.py
@@ -33,7 +33,7 @@ class Gpgme(AutotoolsPackage):
 
     depends_on("gnupg", type="build")
     depends_on("libgpg-error", type="build")
-    depends_on("libassuan@2.4.2:2", type="build")
+    depends_on("libassuan@2.4.2:2", type=("build", "link"))
 
     @classmethod
     def determine_version(cls, exe):

